components/centraldashboard/app/k8s_service.ts

```diff
@@ -24,12 +72,71 @@ export class KubernetesService {
   /** Retrieves the list of events for the given Namespace from the Cluster. */
   async getEventsForNamespace(namespace: string): Promise<k8s.V1Event[]> {
     try {
-      const {body} = await this.k8sApi.listNamespacedEvent(namespace);
+      const {body} = await this.coreAPI.listNamespacedEvent(namespace);
       return body.items;
     } catch (err) {
       console.error(
           `Unable to fetch Events for ${namespace}:`, err.body || err);
       return [];
     }
   }
+
+  /**
+   * Obtains cloud platform information from cluster Nodes,
+   * as well as the Kubeflow version from the Application custom resource.
+   */
+  async getPlatformInfo(): Promise<PlatformInfo> {
+    try {
+      const [provider, version] =
+          await Promise.all([this.getProvider(), this.getKubeflowVersion()]);
+      return {
+        kubeflowVersion: version,
+        provider,
+        providerName: provider.split(':')[0]
+      };
+    } catch (err) {
+      console.error('Unexpected error', err);
+      throw err;
+    }
+  }
+
+  /**
+   * Returns the provider identifier or 'other://' from the K8s cluster.
+   */
+  private async getProvider(): Promise<string> {
+    let provider = 'other://';
+    try {
+      const {body} = await this.coreAPI.listNode();
+      const foundProvider =
```

Comment:
110 - 114 can be written as:
```js
    provider = body.items.map((n) => n.spec.providerID).find(Boolean) || provider;
```

Reply:
Everything else looks good!

Reply:
Understood. I think this is probably stylistic preference more than anything else which I tend to error on the side of block constructs where possible since to me the intuition here is that if it's not set, we won't make any changes to the variable rather than reassign its value to itself. If you feel strongly about making the change I can do so.